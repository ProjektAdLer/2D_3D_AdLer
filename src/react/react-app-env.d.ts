/// <reference types="react-scripts" />

// Images
declare module '*.jpg';
declare module '*.png';
declare module '*.env';

// 3D types
declare module '*.glb';
declare module '*.stl';

declare module '*.jpeg';
declare module '*.gif';
declare module '*.mp4';
