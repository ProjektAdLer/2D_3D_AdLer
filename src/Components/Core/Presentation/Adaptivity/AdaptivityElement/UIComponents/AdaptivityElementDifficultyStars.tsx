import tailwindMerge from "../../../Utils/TailwindMerge";

import requiredSolvedIcon from "../../../../../../Assets/icons/40-difficulties-adaptivity/diffculties-adaptivity-required-solved-icon.svg";
import requiredUnsolvedIcon from "../../../../../../Assets/icons/40-difficulties-adaptivity/diffculties-adaptivity-required-unsolved-icon.svg";
import notRequiredSolvedIcon from "../../../../../../Assets/icons/40-difficulties-adaptivity/diffculties-adaptivity-solved-icon.svg";
import notRequiredUnsolvedIcon from "../../../../../../Assets/icons/40-difficulties-adaptivity/diffculties-adaptivity-unsolved-icon.svg";

export enum AdaptivityElementDifficultyStarState {
  Empty,
  RequiredSolved,
  RequiredTried,
  RequiredUnsolved,
  NotRequiredSolved,
  NotRequiredTried,
  NotRequiredUnsolved,
}

enum AdaptivityElementQuestionDifficulty {
  Easy,
  Medium,
  Hard,
}

// TODO: add more star icons here
function getIcon(
  difficulty: AdaptivityElementQuestionDifficulty,
  status: AdaptivityElementDifficultyStarState
): string | undefined {
  switch (difficulty) {
    case AdaptivityElementQuestionDifficulty.Easy:
      switch (status) {
        case AdaptivityElementDifficultyStarState.RequiredSolved:
          return requiredSolvedIcon;
        case AdaptivityElementDifficultyStarState.RequiredUnsolved:
          return requiredUnsolvedIcon;
        case AdaptivityElementDifficultyStarState.RequiredTried:
          return "";
        case AdaptivityElementDifficultyStarState.NotRequiredSolved:
          return notRequiredSolvedIcon;
        case AdaptivityElementDifficultyStarState.NotRequiredUnsolved:
          return notRequiredUnsolvedIcon;
        case AdaptivityElementDifficultyStarState.NotRequiredTried:
          return "";
      }
    case AdaptivityElementQuestionDifficulty.Medium:
      switch (status) {
        case AdaptivityElementDifficultyStarState.RequiredSolved:
          return requiredSolvedIcon;
        case AdaptivityElementDifficultyStarState.RequiredUnsolved:
          return requiredUnsolvedIcon;
        case AdaptivityElementDifficultyStarState.RequiredTried:
          return "";
        case AdaptivityElementDifficultyStarState.NotRequiredSolved:
          return notRequiredSolvedIcon;
        case AdaptivityElementDifficultyStarState.NotRequiredUnsolved:
          return notRequiredUnsolvedIcon;
        case AdaptivityElementDifficultyStarState.NotRequiredTried:
          return "";
      }
    case AdaptivityElementQuestionDifficulty.Hard:
      switch (status) {
        case AdaptivityElementDifficultyStarState.RequiredSolved:
          return requiredSolvedIcon;
        case AdaptivityElementDifficultyStarState.RequiredUnsolved:
          return requiredUnsolvedIcon;
        case AdaptivityElementDifficultyStarState.RequiredTried:
          return "";
        case AdaptivityElementDifficultyStarState.NotRequiredSolved:
          return notRequiredSolvedIcon;
        case AdaptivityElementDifficultyStarState.NotRequiredUnsolved:
          return notRequiredUnsolvedIcon;
        case AdaptivityElementDifficultyStarState.NotRequiredTried:
          return "";
      }
  }
}

function EmptyStarSlot({ className }: { className: string }): JSX.Element {
  return <div className={tailwindMerge(`${className}`)} />;
}

export default function AdaptivityElementDifficultyStars({
  easyState,
  mediumState,
  hardState,
  starClassName,
}: {
  easyState: AdaptivityElementDifficultyStarState;
  mediumState: AdaptivityElementDifficultyStarState;
  hardState: AdaptivityElementDifficultyStarState;
  starClassName: string;
}) {
  return (
    <div className="flex flex-row h-full">
      {easyState !== AdaptivityElementDifficultyStarState.Empty ? (
        <img
          className={tailwindMerge(`${starClassName}`)}
          alt=""
          src={getIcon(AdaptivityElementQuestionDifficulty.Easy, easyState)}
        />
      ) : (
        <EmptyStarSlot className={starClassName} />
      )}
      {mediumState !== AdaptivityElementDifficultyStarState.Empty ? (
        <img
          className={tailwindMerge(`${starClassName}`)}
          alt=""
          src={getIcon(AdaptivityElementQuestionDifficulty.Medium, mediumState)}
        />
      ) : (
        <EmptyStarSlot className={starClassName} />
      )}
      {hardState !== AdaptivityElementDifficultyStarState.Empty ? (
        <img
          className={tailwindMerge(`${starClassName}`)}
          alt=""
          src={getIcon(AdaptivityElementQuestionDifficulty.Hard, hardState)}
        />
      ) : (
        <EmptyStarSlot className={starClassName} />
      )}
    </div>
  );
}
