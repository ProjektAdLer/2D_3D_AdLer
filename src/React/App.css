body {
  margin: 0;
  height: 100vh;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

div{
  width: 100%;
  height: 100%;
}

section{
  display: inline;
  position: fixed;
  top: 0;
  left: 0;
  width: 120px;
  height: 100vh;
  background-color: #fff;
}

button{
  width: 100px;
  margin: 10px;
  z-index: 5;
  position: fixed;
  left: 0;
  top: 0;
}

canvas {
  width: calc(100% - 100px);
  height: 100%;
  margin-left: 100px;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

.modal {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
}

.modal-content {
  width: 1000px;
  background-color: #fff;
}

.modal-header,
.modal-footer {
  padding: 10px;
  height: 60px;
}

.modal-title {
  margin: 0;
}

.modal-body {
  padding: 10px;
  border-top: 1px solid #eee;
  border-bottom: 1px solid #eee;
}
