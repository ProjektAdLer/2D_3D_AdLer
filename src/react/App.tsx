import React from 'react';
import './App.css';
import BabylonCanvas from './components/BabylonCanvas';

function App() {
	return (
		<React.Fragment>
			<BabylonCanvas style={{ width: '100%', position: 'absolute' }} />
		</React.Fragment>
	);
}

export default App;
