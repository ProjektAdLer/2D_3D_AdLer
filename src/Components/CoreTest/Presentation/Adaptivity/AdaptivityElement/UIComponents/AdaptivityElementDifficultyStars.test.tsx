import { render } from "@testing-library/react";
import React from "react";
import AdaptivityElementDifficultyStars, {
  AdaptivityElementDifficultyStarState,
} from "../../../../../Core/Presentation/Adaptivity/AdaptivityElement/UIComponents/AdaptivityElementDifficultyStars";

describe("AdaptivityElementDifficultyStars", () => {
  test.each([
    AdaptivityElementDifficultyStarState.Empty,
    AdaptivityElementDifficultyStarState.RequiredSolved,
    AdaptivityElementDifficultyStarState.RequiredUnsolved,
    AdaptivityElementDifficultyStarState.RequiredTried,
    AdaptivityElementDifficultyStarState.NotRequiredSolved,
    AdaptivityElementDifficultyStarState.NotRequiredUnsolved,
    AdaptivityElementDifficultyStarState.NotRequiredTried,
  ])("should render with %p for each difficulty", (state) => {
    const { container } = render(
      <AdaptivityElementDifficultyStars
        easyState={state}
        mediumState={state}
        hardState={state}
        starClassName=""
      />
    );
    expect(container).toMatchSnapshot();
  });
});
