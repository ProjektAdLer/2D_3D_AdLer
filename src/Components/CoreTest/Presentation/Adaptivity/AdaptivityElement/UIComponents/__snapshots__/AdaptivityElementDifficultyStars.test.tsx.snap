// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdaptivityElementDifficultyStars should render with 0 for each difficulty 1`] = `
<div>
  <div
    class="flex flex-row h-full"
  >
    <div
      class=""
    />
    <div
      class=""
    />
    <div
      class=""
    />
  </div>
</div>
`;

exports[`AdaptivityElementDifficultyStars should render with 1 for each difficulty 1`] = `
<div>
  <div
    class="flex flex-row h-full"
  >
    <img
      alt=""
      class=""
      src="diffculties-adaptivity-required-solved-icon.svg"
    />
    <img
      alt=""
      class=""
      src="diffculties-adaptivity-required-solved-icon.svg"
    />
    <img
      alt=""
      class=""
      src="diffculties-adaptivity-required-solved-icon.svg"
    />
  </div>
</div>
`;

exports[`AdaptivityElementDifficultyStars should render with 2 for each difficulty 1`] = `
<div>
  <div
    class="flex flex-row h-full"
  >
    <img
      alt=""
      class=""
      src=""
    />
    <img
      alt=""
      class=""
      src=""
    />
    <img
      alt=""
      class=""
      src=""
    />
  </div>
</div>
`;

exports[`AdaptivityElementDifficultyStars should render with 3 for each difficulty 1`] = `
<div>
  <div
    class="flex flex-row h-full"
  >
    <img
      alt=""
      class=""
      src="diffculties-adaptivity-required-unsolved-icon.svg"
    />
    <img
      alt=""
      class=""
      src="diffculties-adaptivity-required-unsolved-icon.svg"
    />
    <img
      alt=""
      class=""
      src="diffculties-adaptivity-required-unsolved-icon.svg"
    />
  </div>
</div>
`;

exports[`AdaptivityElementDifficultyStars should render with 4 for each difficulty 1`] = `
<div>
  <div
    class="flex flex-row h-full"
  >
    <img
      alt=""
      class=""
      src="diffculties-adaptivity-solved-icon.svg"
    />
    <img
      alt=""
      class=""
      src="diffculties-adaptivity-solved-icon.svg"
    />
    <img
      alt=""
      class=""
      src="diffculties-adaptivity-solved-icon.svg"
    />
  </div>
</div>
`;

exports[`AdaptivityElementDifficultyStars should render with 5 for each difficulty 1`] = `
<div>
  <div
    class="flex flex-row h-full"
  >
    <img
      alt=""
      class=""
      src=""
    />
    <img
      alt=""
      class=""
      src=""
    />
    <img
      alt=""
      class=""
      src=""
    />
  </div>
</div>
`;

exports[`AdaptivityElementDifficultyStars should render with 6 for each difficulty 1`] = `
<div>
  <div
    class="flex flex-row h-full"
  >
    <img
      alt=""
      class=""
      src="diffculties-adaptivity-unsolved-icon.svg"
    />
    <img
      alt=""
      class=""
      src="diffculties-adaptivity-unsolved-icon.svg"
    />
    <img
      alt=""
      class=""
      src="diffculties-adaptivity-unsolved-icon.svg"
    />
  </div>
</div>
`;
